tests/test_project/tests/test_menu.py

```diff
@@ -0,0 +1,284 @@
+from unittest.mock import patch
+
+from django.apps import registry
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ImproperlyConfigured
+from django.test import RequestFactory, TestCase
+from django.urls import reverse
+from openwisp_utils.admin_theme.menu import (
+    MenuGroup,
+    MenuItem,
+    MenuLink,
+    register_menu_groups,
+)
+from openwisp_utils.utils import SortedOrderedDict
+
+
+class TestMenuSchema(TestCase):
```

Comment:
The tests here seem too big to me, each test is testing a lot of things.
I think having smaller tests is beneficial because
1. they can run in parallel
2. If there a failure in step X another case which is on step X+1 might not be reported. 
3. We can be sure that data is pristine in smaller tests. In larger tests, it's possible to mistakenly taint data.

However, I'd like to know what others thing about it; hence I'd wait for others reviews before recommending this change!

Reply:
I think the size of the test suite is still fine and does not need to be split.
In general, let's try to keep test suites under 1K lines.

> they can run in parallel

Yeah but when not running in parallel, processing more test classes can take more time because the DB is destroyed and recreated each time. So each approach has pros and cons, the right thing to do is to keep it simple do additional steps only when pain makes it really needed.

> If there a failure in step X another case which is on step X+1 might not be reported.

If one subtests fails the others will be executed anyway. 

> We can be sure that data is pristine in smaller tests. In larger tests, it's possible to mistakenly taint data.

Subtests should be used only to perform tests which are related to a specific feature, which I think is what Nitesh is doing here, so I think that's very good.